pkg/virt-handler/vm.go

```diff
@@ -557,6 +557,12 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 
 			existingInterfacesSpecByName := map[string]v1.Interface{}
 			for _, existingInterfaceSpec := range vmi.Spec.Domain.Devices.Interfaces {
+				if existingInterfaceSpec.MacAddress != "" {
+					if domainInterfaceStatus, ok := domainInterfaceStatusByMac[existingInterfaceSpec.MacAddress]; ok {
+						domainInterfaceStatus.Name = existingInterfaceSpec.Name
+						domainInterfaceStatusByMac[domainInterfaceStatus.Mac] = domainInterfaceStatus
```

Comment:
shouldn't this line be added to the former loop, that specifically addresses `domainInterfaceStatusByMac`?

Reply:
if we do it there, it will be O(N*M) to find the right one (using range on all entries)
here its amortized O(1)

Reply:
you're totally right.